modules/runner/src/integrationTest/java/org/apache/ignite/internal/runner/app/client/ItThinClientCriteriaQueryTest.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.runner.app.client;
+
+import static org.apache.ignite.table.criteria.CriteriaQueryOptions.builder;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.hasProperty;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import com.google.common.collect.Lists;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Map;
+import org.apache.ignite.Ignite;
+import org.apache.ignite.internal.testframework.WorkDirectory;
+import org.apache.ignite.sql.async.AsyncClosableCursor;
+import org.apache.ignite.table.RecordView;
+import org.apache.ignite.table.Tuple;
+import org.hamcrest.FeatureMatcher;
+import org.hamcrest.Matcher;
+import org.jetbrains.annotations.Nullable;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInfo;
+
+/**
+ * Thin client criteria query test.
+ */
+@SuppressWarnings("resource")
+public class ItThinClientCriteriaQueryTest extends ItAbstractThinClientTest {
+    /** {@inheritDoc} */
+    @Override
+    @BeforeAll
+    void beforeAll(TestInfo testInfo, @WorkDirectory Path workDir) throws InterruptedException {
+        super.beforeAll(testInfo, workDir);
+
+        populateData(client(), TABLE_NAME);
+    }
+
+    @Test
+    public void testBasicQueryCriteriaRecordBinaryView() {
+        RecordView<Tuple> view = client().tables().table(TABLE_NAME).recordView();
+
+        List<Tuple> res = Lists.newArrayList(view.queryCriteria(null, null));
+        assertThat(res, containsInAnyOrder(
+                tupleValue(COLUMN_KEY, is(0)),
+                tupleValue(COLUMN_KEY, is(1)),
+                tupleValue(COLUMN_KEY, is(2))
+        ));
+
+        AsyncClosableCursor<Tuple> ars = view.queryCriteriaAsync(null, null, builder().pageSize(2).build()).join();
```

Comment:
it would be nice to close cursor at the end each tests.

Reply:
Added close cursor.